filecoin-proofs/src/bin/phase2.rs

```diff
@@ -355,492 +436,664 @@ fn create_initial_params<Tree: 'static + MerkleTreeTrait>(
+    let mut streamer = if param_size.is_large() {
+        Streamer::new_from_large_file(path_before, read_raw, write_raw).unwrap_or_else(|e| {
+            panic!(
+                "failed to make streamer from large `{}`: {}",
+                path_before, e
+            );
+        })
+    } else {
+        Streamer::new(path_before, read_raw, write_raw).unwrap_or_else(|e| {
+            panic!(
+                "failed to make streamer from small `{}`: {}",
+                path_before, e
+            );
+        })
+    };
+
+    info!("writing small 'after' params to file: {}", path_after);
+    let file_after = File::create(&path_after).unwrap_or_else(|e| {
+        panic!(
+            "failed to create 'after' params file `{}`: {}",
+            path_after, e
+        );
+    });
+
+    let contrib = streamer
+        .contribute(&mut rng, file_after, CHUNK_SIZE)
+        .expect("failed to make contribution");
+
+    let contrib_str = hex_string(&contrib);
     info!(
-        "successfully made contribution, contribution hash: {}, dt_contribute={}s",
-        hex::encode(&contribution[..]),
-        start.elapsed().as_secs()
+        "successfully made contribution: {}, dt_contribute={}s",
+        contrib_str,
+        start_contrib.elapsed().as_secs()
     );
 
-    let path_after = params_filename(proof, hasher, sector_size, &head, param_number_before + 1);
-    info!("writing 'after' params to file: {}", path_after);
-    let file_after = File::create(path_after).unwrap();
-    let mut params_writer = BufWriter::with_capacity(1024 * 1024, file_after);
-    params.write(&mut params_writer).unwrap();
+    let contrib_path = format!("{}.contrib", path_after);
+    info!("writing contribution hash to file: {}", contrib_path);
+    fs::write(&contrib_path, contrib_str).unwrap_or_else(|e| {
+        panic!(
+            "failed to write contribution to file `{}`: {}",
+            contrib_path, e
+        );
+    });
+
     info!(
         "successfully made contribution, dt_total={}s",
         start_total.elapsed().as_secs()
     );
 }
 
-/// Verifies a sequence of parameter transitions against a sequence of corresponding contribution
-/// hashes. For example, verifies that the first digest in `contribution_hashes` transitions the
-/// first parameters file in `param_paths` to the second file, then verifies that the second
-/// contribution hash transitions the second parameters file to the third file.
-fn verify_param_transitions(param_paths: &[&str], contribution_hashes: &[[u8; 64]]) {
-    let start_total = Instant::now();
+fn convert_small<'a>(path_before: &'a str) {
+    let (proof, hasher, sector_size, head, param_number, param_size, read_raw) =
+        parse_params_filename(path_before);
 
-    assert_eq!(
-        param_paths.len() - 1,
-        contribution_hashes.len(),
-        "the number of contributions must be one less than the number of parameter files"
+    let write_raw = !read_raw;
+
+    info!(
```

Comment:
Could you please update this log

```rust
info!(
    "copying params into raw format for circuit: {proof}-{hasher}-{sector_size}-{head}-{num}-{param_size}",
    proof = proof.pretty_print(),
    hasher = hasher.pretty_print(),
    sector_size = sector_size.pretty_print(),
    head = head,
    num = param_number,
    param_size = param_size.pretty_print()
);
```

Reply:
Done, but with modification to indicate whether we are converting to or from raw.